lib/metasploit/framework/login_scanner/base.rb

```diff
@@ -251,8 +251,13 @@ def scan!
                   break if total_error_count >= 10
                 end
               end
-            rescue => e
-              elog('Attempt may not yield a result', error: e)
+            rescue Rex::ConnectionRefused, Rex::SocketError, Rex::HostCommunicationError, Rex::ConnectionError, RuntimeError, NoMethodError, TypeError, Rex::TimeoutError, OpenSSL::Cipher::CipherError, Rex::ConnectionProxyError, Errno::ECONNRESET, Errno::EINTR, Errno::EPIPE, ::SystemCallError, ::EOFError, Errno::ENOTCONN, ::Timeout::Error, Rex::Proto::Kerberos::Model::Error::KerberosEncryptionNotSupported, ::JSON::ParserError, Errno::ECONNRESET, OpenSSL::SSL::SSLError, Rex::StreamClosedError, ::Rex::Proto::Kerberos::Model::Error::KerberosError, Errno::ETIMEDOUT, StandardError => e
```

Comment:
Some of these exceptions are definitely not login related issues, and are more likely exceptions raised by broken modules with Ruby issues - i.e. `NoMethodError` and `RuntimeError` and `TypeError`

Likewise `rescue StandardError => e` is just the same as `rescue => e` ([details](https://thoughtbot.com/blog/rescue-standarderror-not-exception)) - I think these code changes might need some more cycles spent on analysing the approach here 🤔

Reply:
Good call, I'll take another pass 🫡